spec/lib/message_bus/client_spec.rb

```diff
@@ -359,35 +365,45 @@ def parse_chunk(data)
       end
 
       describe 'when MessageBus#client_message_filters has been configured' do
-
         it 'filters messages correctly' do
           message = MessageBus::Message.new(1, 2, '/test/5', 'hello')
           @client.allowed?(message).must_equal(true)
 
-          @bus.register_client_message_filter('/test') do |m|
+          @bus.register_client_message_filter('/test') do |_, m|
             m.data != 'hello'
           end
 
+          @client = setup_client('abc')
+
           @client.allowed?(message).must_equal(false)
         end
 
         it 'filters messages correctly when multiple filters have been configured' do
-
           bob_message = MessageBus::Message.new(1, 2, '/test/5', 'bob')
           fred_message = MessageBus::Message.new(1, 2, '/test/5', 'fred')
+          foo_message = MessageBus::Message.new(1, 2, '/footest/5', 'foo')
           random_message = MessageBus::Message.new(1, 2, '/test/77', 'random')
 
-          @bus.register_client_message_filter('/test') do |message|
+          @params = { 't' => 1 }
+
+          @bus.register_client_message_filter('/test') do |_, message|
             message.data == 'bob' || message.data == 'fred'
           end
 
-          @bus.register_client_message_filter('/test') do |message|
+          @bus.register_client_message_filter('/test') do |_, message|
```

Comment:
Hm.. interesting registering two filters for the same channel?

Reply:
According to the docs - multiple filters can be registered for the same channel. Btw, that is not a channel name, but channel **prefix**. We can also provide a regexp there.